### ISSUE TITLE
json2xml not working as intended.

### ISSUE THREAD

**Issue by DiglidiDudeNG** [nashwaan/xml-js#6]
To reproduce, I've first made an XML file that looks similar to what I'm looking for : 

```xml
<ZohoCreator>
    <applicationslist>
        <application name="testapp">
            <formlist>
                <form name="Untitled_Form">
                    <add>
                        <field name="Subform_Single_Line">
                            <value>Value1</value>
                        </field>
                    </add>
                </form>
                <form name="Untitled_Form">
                    <add>
                        <field name="Subform_Single_Line">
                            <value>Value2</value>
                        </field>
                    </add>
                </form>
            </formlist>
        </application>
        <application name="derp">
            <formlist>
            </formlist>
        </application>
    </applicationslist>
</ZohoCreator>
```

Which gave me this : 

```json
{
    "ZohoCreator": {
        "applicationslist": {
            "application": [
                {
                    "_attributes": {
                        "name": "testapp"
                    },
                    "formlist": {
                        "form": [
                            {
                                "_attributes": {
                                    "name": "Untitled_Form"
                                },
                                "add": {
                                    "field": {
                                        "_attributes": {
                                            "name": "Subform_Single_Line"
                                        },
                                        "value": {
                                            "_text": "Value1"
                                        }
                                    }
                                }
                            },
                            {
                                "_attributes": {
                                    "name": "Untitled_Form"
                                },
                                "add": {
                                    "field": {
                                        "_attributes": {
                                            "name": "Subform_Single_Line"
                                        },
                                        "value": {
                                            "_text": "Value2"
                                        }
                                    }
                                }
                            }
                        ]
                    }
                },
                {
                    "_attributes": {
                        "name": "derp"
                    },
                    "formlist": {}
                }
            ]
        }
    }
}
```

So far so good. Now that I've got a well-formatted json according to your library's needs, let's try parsing it again, but backwards this time!

```xml
<ZohoCreator>
    <applicationslist>
        <application>
            <0 name="testapp">
                <formlist>
                    <form>
                        <0 name="Untitled_Form">
                            <add>
                                <field name="Subform_Single_Line">
                                    <value>Value1</value>
                                </field>
                            </add>
                        </0>
                        <1 name="Untitled_Form">
                            <add>
                                <field name="Subform_Single_Line">
                                    <value>Value2</value>
                                </field>
                            </add>
                        </1>
                    </form>
                </formlist>
            </0>
            <1 name="derp">
                <formlist/>
            </1>
        </application>
    </applicationslist>
</ZohoCreator>
```

Yeah. It's nothing like what we got. This is kinda killing it for me, as I was trying to convert a JSON into an XML in a easier fashion.

**Comment by nashwaan** [comment]
Thanks for reporting this issue.
This is now fixed in v0.9.7.
Let me know if the problem still persists.

**Comment by nashwaan** [comment]
@DiglidiDudeNG 
Closing this issue as no more problem reported. Will re-open if the fix is not adequate.
